Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -228,10 +267,75 @@ public Task<TResult> InvokeWithParameterObjectAsync<TResult>(string targetName,
 
         private void OnSettingsChanged(object sender, EventArgs e) => SendDidChangeConfigurations().DoNotWait();
 
+        private void OnAnalysisComplete(object sender, EventArgs e) {
+            // Used by test code to know when it's okay to try and use intellisense
+            _readyTcs.TrySetResult(0);
+        }
+
         private Task SendDidChangeConfigurations() {
             return Task.WhenAll(_clientContexts.Select(c => SendDidChangeConfiguration(c)));
         }
 
+        private async Task SendDocumentOpensAsync() {
+            // This should be handled by the VSSDK if they ever support workspace folder change notifications
+            // Meaning we shouldn't need to do this if they do it for us.
+            await JoinableTaskContext.Factory.SwitchToMainThreadAsync();
+            IComponentModel componentModel = Site.GetService(typeof(SComponentModel)) as IComponentModel;
+            Assumes.Present(componentModel);
+
+            SVsServiceProvider syncServiceProvider = componentModel.GetService<SVsServiceProvider>();
+            RunningDocumentTable rdt = new RunningDocumentTable(syncServiceProvider);
+
+            foreach (RunningDocumentInfo info in rdt) {
+                if (this.TryGetOpenedDocumentData(info, out ITextBuffer textBuffer, out string filePath)
+                    && textBuffer != null
+                    && textBuffer.IsPythonContent()) {
+
+                    var textDocumentItem = new TextDocumentItem {
+                        Uri = new Uri(filePath),
+                        Version = textBuffer.CurrentSnapshot.Version.VersionNumber,
+                        LanguageId = textBuffer.ContentType.DisplayName,
+                    };
+
+                    var param = new DidOpenTextDocumentParams {
+                        TextDocument = textDocumentItem,
+                    };
+                    param.TextDocument.Text = textBuffer.CurrentSnapshot.GetText();
+
+                    await InvokeTextDocumentDidOpenAsync(param).ConfigureAwait(false);
+                }
+            }
+        }
```

Comment:
should we group these tasks in a `Task.WhenAll`

Reply:
Good idea. I don't see why they can't all go at once.